addons/account/models/account_tax.py

```diff
@@ -783,7 +783,11 @@ def add_extra_base(other_batch, tax_data, sign):
                 # On a price unit of 120, t1 is computed first since the tax amount affects the price unit.
                 # Then, t2 can be computed on 120 + 1 = 121.
                 # However, since t1 is not price-included, its base amount is computed by removing first the tax amount of t2.
-                if not special_mode:
+                if not special_mode or special_mode == 'total_included':
+                    if not batch['include_base_amount']:
+                        for other_batch in batches_after:
+                            if other_batch['_original_price_include']:
+                                add_extra_base(other_batch, tax_data, -1)
```

Comment:
We should also add some comments here why we're removing some of the tax amount from `batches_after` here